src/crawlee/beautifulsoup_crawler/_beautifulsoup_crawler.py

```diff
@@ -191,6 +197,9 @@ async def enqueue_links(
                 if label is not None:
                     link_user_data.setdefault('label', label)
 
+                data = {'crawlDepth': context.request.crawl_depth + 1}
+                link_user_data.setdefault('__crawlee', CrawleeRequestData(**data))
```

Comment:
```suggestion
                link_user_data.crawlee = link_user_data.crawlee or CrawleeRequestData()
                link_user_data.crawlee.crawl_depth = context.request.crawl_depth + 1
```

I'd prefer not fiddling with field aliases if possible.